module/VuFind/src/VuFind/Db/DbBuilder.php

```diff
@@ -0,0 +1,235 @@
+     * @param string   $rootUser      Root username for connecting to database
+     * @param string   $rootPass      Root password for connecting to database
+     * @param bool     $returnSqlOnly Set to true to return SQL without actually manipulating the database
+     * @param string[] $steps         Array of steps to run (legal values: pre, main, post); omit for all steps
+     *
+     * @return string
+     * @throws Exception
+     * @throws DBALException
+     */
+    public function build(
+        string $newName,
+        string $newUser,
+        string $newPass,
+        string $driver = 'mysql',
+        string $dbHost = 'localhost',
+        string $vufindHost = 'localhost',
+        string $rootUser = 'root',
+        string $rootPass = '',
+        bool $returnSqlOnly = false,
+        array $steps = []
+    ): string {
+        try {
+            // We need a default database name to use to establish a connection:
+            $dbName = ($driver == 'pgsql') ? 'template1' : 'mysql';
+            $connectionParams = [
+                'driver' => $this->dbFactory->getDriverName($driver),
+                'host' => $dbHost,
+                'user' => $rootUser,
+                'password' => $rootPass,
+            ];
+            $db = $this->dbFactory->getConnectionFromOptions(
+                $connectionParams + ['dbname' => $dbName]
+            );
+        } catch (\Exception $e) {
+            throw new \Exception(
+                'Problem initializing database adapter; '
+                . 'check for missing ' . $driver
+                . ' library. Details: ' . $e->getMessage(),
+                'error',
+                $e
+            );
+        }
+        // Invert the steps list into a list of steps we should skip (no skipping if empty list):
+        $allSteps = ['pre', 'main', 'post'];
+        $skip = $steps ? array_diff($allSteps, $steps) : [];
+
+        // Get SQL together
+        $escapedPass = $returnSqlOnly
+            ? "'" . addslashes($newPass) . "'"
+            : $db->quote($newPass);
+        $preCommands = in_array('pre', $skip)
+            ? [] : $this->getPreCommands($driver, $newName, $vufindHost, $newUser, $escapedPass);
+        $sql = in_array('main', $skip) ? '' : $this->getMainSql($driver);
+        $postCommands = in_array('post', $skip)
+            ? [] : $this->getPostCommands($driver, $newUser);
+        $omnisql = '';
+        foreach ($preCommands as $query) {
+            $omnisql .= $query . ";\n";
+            if (!$returnSqlOnly) {
+                $db->executeQuery($query);
+            }
+        }
+        if ($sql) {
+            $omnisql .= "\n" . $sql . "\n";
+            if (!$returnSqlOnly) {
+                $db = $this->dbFactory->getConnectionFromOptions(
+                    $connectionParams + ['dbname' => $newName]
+                );
+                $statements = explode(';', $sql);
```

Comment:
Should we split this with a regex here as well? Maybe this could be done in a helper method somewhere to be used in #4577 as well?
```suggestion
                $statements = preg_split('/;\s*([\r\n]|$)/', $sql);
```

Reply:
I'll accept your suggestion as-is for now and refactor when I merge with #4577. Thanks!